### PR TITLE
Add test, debug and build section

### PR DESCRIPTION
**Pull request by SueChaplain**
Draft test and debug sections included for the nodejs-express
simple template.

Build section also added.

Signed-off-by: Sue Chaplain <sue_chaplain@uk.ibm.com>

### CODE REVIEW

**Comment by SueChaplain** [comment]
@helenmasters 

Added your test and debug sections with some extra system output (which helped me understand what happens). I jigged things about a little and brought in some Kabanero-specific terminology. Please can you check that I haven't inadvertently mis-quoted anything. Think you should be able to merge. I'll add your `build` input tomorrow morning and all being well we can ask for a final review from Alan and Nate.

Thank you!

**Comment by SueChaplain** [comment]
@helenmasters - found time to add build. Please review. 

Just noticed that I have inadvertently flipped test and debug compared to your input (!) ... can reverse if the order makes more sense.

File: README.adoc [hunk 1]
```diff
@@ -58,13 +59,17 @@ to create, run, test, and deploy an application built from a Kabanero Collection
 
 == Discovering Appsody repositories and stacks
 
-Appsody is an intergal component of Kabanero, its designed to enable applications to be built and tested inside a docker container, which then can be deployed by Kabanero into a target host environment.  
+Appsody is an integral component of Kabanero, designed to enable applications to be built and tested inside a docker container, which can then
```
Comment (helenmasters) [inline README.adoc]: I personally don't like the phrase "Appsody is an integral component of Kabanero" as you could read this as us defining what Appsody is and yet Appsody has no dependency on Kabanero.

docker container -> Docker container

File: README.adoc [hunk 2]
```diff
@@ -76,23 +81,26 @@ NAME            URL
...
 
-Run the following command to add the latest public Kabanero Collections repository (it is recommended that the installer of the Kabanero runtime, make a copy of the collection hub and provide that to you, so that you are not suprised when the public collection hub changes)
+In this example, you will add the latest *public* Kabanero Collection Hub repository instead of a cloned Kabanero
+Collection Hub repository that has been customised for your organisation. Run the following command:
 
 ```
 appsody repo add kabanero https://github.com/kabanero-io/collections/releases/latest/download/kabanero-index.yaml
 ```
 
+NOTE: The *public* Kabanero Collection Hub is updated on a regular basis. By using a clone, you avoid adopting unplanned changes.
```
Comment (helenmasters) [inline README.adoc]: The phrase "By using a clone, ..." makes me think that we are using a clone in this case but we're not, this example shows us using the *public* Kabanero Collection Hub so I find that wording confusing.

File: README.adoc [hunk 3]
```diff
@@ -150,11 +158,11 @@ kabanero      nodejs-express           0.2.5      *simple, skaffold   Express we
 kabanero      nodejs-loopback          0.1.4      *scaffold           LoopBack 4 API Framework for Node.js
 ----
 
-In the output you can see multiple stacks (IDs) in the `kabanero` collection hub. Each collection includes a version number,
+In the output you can see multiple collections (IDs) in the `kabanero` Collection Hub. Each collection includes a version number,
 one or more templates (the asterisk (*****) indicates the default template), and a description.
```
Comment (helenmasters) [inline README.adoc]: why are there 5 asterisks displayed in brackets after the word asterisk?

Comment (SueChaplain) [reply README.adoc]: Not sure where these came from. I've modified to say 

.. (an asterisk (*) indicates the default template) ...

File: README.adoc (hunk 2 above)
Comment (SueChaplain) [reply README.adoc]: Changed the note and moved it above **Run the following command**

NOTE: The *public* Kabanero Collection Hub is updated on a regular basis. By making a copy of the public hub you avoid adopting unplanned changes.

File: README.adoc (hunk 1 above)
Comment (SueChaplain) [reply README.adoc]: How about:

"Kabanero contains Appsody, which is designed to enable applications to be built and tested inside a Docker container. When built, the Docker container can be deployed by Kabanero into a target host environment"

@ralanlittle - modification to your input, so please comment.

**Review by helenmasters — APPROVED** [review]
lgtm

**Comment by SueChaplain** [comment]
The changes are difficult to review in this PR. Merging so that Alan and Nate can do an easier review off the master.

**Pull request merged**